curvefs/src/client/space_client.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+ *  Copyright (c) 2021 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+
+/*
+ * Project: curve
+ * Created Date: Thur Jun 17 2021
+ * Author: lixiaocui
+ */
+
+
+#include "curvefs/src/client/space_client.h"
+
+namespace curvefs {
+namespace client {
+
+CURVEFS_ERROR
+SpaceAllocServerClientImpl::SpaceAllocRPCExcutor::DoRPCTask(RPCFunc task) {
+    brpc::Channel channel;
+    int ret = channel.Init(opt_.spaceaddr.c_str(), nullptr);
+    if (ret != 0) {
+        LOG(WARNING) << "Init channel failed, addr = " << opt_.spaceaddr;
+        return static_cast<CURVEFS_ERROR>(-EHOSTDOWN);
+    }
+
+    brpc::Controller cntl;
+    cntl.set_timeout_ms(opt_.rpcTimeoutMs);
+
+    ret = task(&channel, &cntl);
+    return static_cast<CURVEFS_ERROR>(ret);
+}
+
+CURVEFS_ERROR
+SpaceAllocServerClientImpl::Init(const SpaceAllocServerOption &spaceopt,
+                                 SpaceBaseClient *baseclient) {
+    basecli_ = baseclient;
+    excutor_.SetOption(spaceopt);
+    return CURVEFS_ERROR::OK;
+}
+
+#define RPCTaskDefine                                                          \
+    [&](brpc::Channel * channel, brpc::Controller * cntl) -> int
```

Comment:
Why not use CURVEFS_ERROR as return code to invoid type cast.

Reply:
define  CURVEFS_ERROR as return will cause compile error